test/Microsoft.TestPlatform.Common.UnitTests/Filtering/FastFilterTests.cs

```diff
@@ -90,6 +90,20 @@ public void FastFilterWithSingleEqualsClause()
         Assert.IsFalse(fastFilter.Evaluate(s => "Test2"));
     }
 
+    [TestMethod]
+    public void AvoidingStackOverFlow()
+    {
+        var testCaseFilter = "Test1";
+        for (int i = 0; i < 1e5; i++)
+        {
+            testCaseFilter += "|Test2";
+        }
```

Comment:
That doesn't matter much for tests but in general you should consider `+=` string concatenation inside a loop as a bug because you will cause a lot of instantiations. In such case, use a `StringBuilder` and when you are done call `ToString` on it to create your string.